That didn't take long to patch, unfortunately it doesn't seem to build with Chroot enabled due to networking being disabled.

---

Reviewed 1 of 1 files at r1.
Review status: all files reviewed at latest revision, 1 unresolved discussion, some commit checks broke.

---

*[pkgs/os-specific/linux/sysdig/default.nix, line 27 \[r1\]](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-KN7iL1eSNjj9acLFmNG:-KN7iL1eSNjj9acLFmNH:1577229231) ([raw file](https://github.com/nixos/nixpkgs/blob/794b93992d9eee08f6859a6f981478ae06832324/pkgs/os-specific/linux/sysdig/default.nix#L27)):*
> ```Nix
>   cmakeFlags = [
>     "-DUSE_BUNDLED_DEPS=OFF"
>     "-DUSE_BUNDLED_JQ=ON"
> ```

With this enabled, it seems to try and download fromthe internet during the build, which isn't allowed:


```
Scanning dependencies of target jq
[  1%] Creating directories for 'jq'
[  2%] Performing download step (download, verify and extract) for 'jq'
-- downloading...
     src='http://download.draios.com/dependencies/jq-1.5.tar.gz'
     dst='/tmp/nix-build-sysdig-0.11.0.drv-0/sysdig-0.11.0/build/jq-prefix/src/jq-1.5.tar.gz'
     timeout='none'
CMake Error at jq-stamp/download-jq.cmake:27 (message):
  error: downloading 'http://download.draios.com/dependencies/jq-1.5.tar.gz'
  failed

    status_code: 6
    status_string: "Couldn't resolve host name"
    log: getaddrinfo(3) failed for download.draios.com:80

  Couldn't resolve host 'download.draios.com'

  Closing connection 0
```

---


*Comments from [Reviewable](https://reviewable.io:443/reviews/nixos/nixpkgs/17124#-:-KN7hyzOzHRczuZZnfPI:546775244)*
<!-- Sent from Reviewable.io -->

--- a/pkgs/os-specific/linux/sysdig/default.nix
+++ b/pkgs/os-specific/linux/sysdig/default.nix
@@ -1,22 +1,30 @@
-{stdenv, fetchurl, cmake, luajit, kernel, zlib, ncurses, perl, jsoncpp, libb64, openssl, curl}:
+{stdenv, fetchurl, fetchFromGitHub, cmake, luajit, kernel, zlib, ncurses, perl, jsoncpp, libb64, openssl, curl}:
 let
   inherit (stdenv.lib) optional optionalString;
   s = rec {
-    baseName="sysdig";
-    version = "0.10.0";
-    name="${baseName}-${version}";
-    url="https://github.com/draios/sysdig/archive/${version}.tar.gz";
-    sha256 = "0hs0r9z9j7padqdcj69bwx52iw6gvdl0w322qwivpv12j3prcpsj";
+    name = "sysdig-${version}";
+    version = "0.11.0";
+    owner = "draios";
+    repo = "sysdig";
+    rev = version;
+    sha256 = "131bafa7jy16r2jwph50j0bxwqdvr319fsfhqkavx6xy18i31q3v";
   };
   buildInputs = [
     cmake zlib luajit ncurses perl jsoncpp libb64 openssl curl
   ];
+  # sysdig-0.11.0 depends on some headers from jq which are not
+  # installed by default.
+  # Relevant sysdig issue: https://github.com/draios/sysdig/issues/626
+  jq-prefix = fetchurl {
+    url="https://github.com/stedolan/jq/releases/download/jq-1.5/jq-1.5.tar.gz";
+    sha256="0g29kyz4ykasdcrb0zmbrp2jqs9kv1wz9swx849i2d1ncknbzln4";
+  };
 in
 stdenv.mkDerivation {
   inherit (s) name version;
   inherit buildInputs;
-  src = fetchurl {
-    inherit (s) url sha256;
+  src = fetchFromGitHub {
+    inherit (s) owner repo rev sha256;
   };
   postPatch = ''
     sed '1i#include <cmath>' -i userspace/libsinsp/{cursesspectro,filterchecks}.cpp
@@ -24,11 +32,17 @@ stdenv.mkDerivation {
 
   cmakeFlags = [
     "-DUSE_BUNDLED_DEPS=OFF"
+    "-DUSE_BUNDLED_JQ=ON"
+    "-DSYSDIG_VERSION=${s.version}"
   ] ++ optional (kernel == null) "-DBUILD_DRIVER=OFF";
   preConfigure = ''
     export INSTALL_MOD_PATH="$out"
   '' + optionalString (kernel != null) ''
     export KERNELDIR="${kernel.dev}/lib/modules/${kernel.modDirVersion}/build"
+  '';
+  preBuild = ''
+    mkdir -p jq-prefix/src
+    cp ${jq-prefix} jq-prefix/src/jq-1.5.tar.gz
   '';
   postInstall = optionalString (kernel != null) ''
     make install_driver

